app/controllers/sessions_controller.rb

```diff
@@ -8,6 +8,7 @@ def create
     user = User.find_by(email: params[:email])
     if user && user.authenticate(params[:password])
       session[:user_id] = user.id
+      user.update_column(:token, nil) # reset the token column to nil after a password has been reset
```

Comment:
This is not a good solution. User does not have to sign in immediately after he changed password and whoever steals the database can use the token. I recommend to remove this line and set token to nil right when your app sets new password not here.